src/Storage/Adapter/KeyListIterator.php

```diff
@@ -101,15 +102,17 @@ public function setMode($mode)
      */
     public function current()
     {
-        if ($this->mode == IteratorInterface::CURRENT_AS_SELF) {
+        if ($this->mode === IteratorInterface::CURRENT_AS_SELF) {
```

Comment:
These changes require careful checking

Reply:
I did. 
The mode is being casted to integer in `setMode` method and thus supposed to be integer.
Constants are integer aswell. 👍🏼

Reply:
As this will land in a new major release, I keep this in mind when adding strict typehints. Thus, we would be fine.